data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -284,24 +292,35 @@ private void logFailure(final List<FailedBulkOperation> failedBulkOperations, fi
       dlqObjects.forEach(dlqObject -> {
         final FailedDlqData failedDlqData = (FailedDlqData) dlqObject.getFailedData();
         final String message = failure == null ? failedDlqData.getMessage() : failure.getMessage();
+        final EventHandle eventHandle = (EventHandle)dlqObject.getEventHandle();
         try {
           dlqFileWriter.write(String.format("{\"Document\": [%s], \"failure\": %s}\n",
               BulkOperationWriter.dlqObjectToString(dlqObject), message));
+          acknowledgementSetManager.releaseEventReference(eventHandle, true);
         } catch (final IOException e) {
           LOG.error(SENSITIVE, "DLQ failure for Document[{}]", dlqObject.getFailedData(), e);
+          acknowledgementSetManager.releaseEventReference(eventHandle, false);
         }
       });
     } else if (dlqWriter != null) {
       try {
         dlqWriter.write(dlqObjects, pluginSetting.getPipelineName(), pluginSetting.getName());
+        dlqObjects.forEach((dlqObject) -> {
+            final EventHandle eventHandle = (EventHandle)dlqObject.getEventHandle();
+            acknowledgementSetManager.releaseEventReference(eventHandle, true);
+        });
       } catch (final IOException e) {
         dlqObjects.forEach(dlqObject -> {
+          final EventHandle eventHandle = (EventHandle)dlqObject.getEventHandle();
           LOG.error(SENSITIVE, "DLQ failure for Document[{}]", dlqObject.getFailedData(), e);
+          acknowledgementSetManager.releaseEventReference(eventHandle, false);
```

Comment:
The DLQ plugins should implement this logic. Otherwise DLQ's will not honor events with E2E Acks enabled and it will require any plugin author using a DLQ plugin to enforce E2E acks around the DLQ.

Reply:
@cmanning09 We agreed to defer this to a later time.